### PR TITLE
Store committed data in provenance directory if path is not given

### PR DESCRIPTION
**Pull request by apoorvaeternity**
If the user does not provide a custom path to store the committed dataset the committed dataset will be stored in the `PROVENANCE_DIR` taken from the evironment.

### CODE REVIEW

File: retriever/__main__.py [hunk 1]
```diff
@@ -166,7 +166,7 @@ def main():
                         count += 1
             return
         elif args.command == 'commit':
-            commit(dataset=args.dataset, path=os.path.normpath(args.path[0]),
+            commit(dataset=args.dataset, path=os.path.normpath(args.path[0]) if args.path else None,
```
Comment (henrykironde) [inline retriever/__main__.py]: Line too long

File: retriever/lib/defaults.py [hunk 2]
```diff
@@ -34,6 +34,9 @@
 "predicts.py"
 ]
 
+# Provenance directory(to store committed datasets by default)
+PROVENANCE_DIR = os.environ.get('PROVENANCE_DIR', os.path.expanduser('~/.retriever_provenance/'))
```
Comment (henrykironde) [inline retriever/lib/defaults.py]: Could we refactor this

**Comment by henrykironde** [comment]
Thanks @apoorvaeternity for working on this.

**Comment by henrykironde** [comment]
Since the tests are failing I will look into how we could get everything updated with master before we merge.

**Comment by henrykironde** [comment]
Any Idea why this is failing @apoorvaeternity?

**Comment by apoorvaeternity** [comment]
The same reason. We need to update the docker file.

**Comment by apoorvaeternity** [comment]
@henrykironde I have updated the Dockerfile with the `master` branch the build is passing now.

**Pull request merged**